### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by AXeL-dev** [AXeL-dev/distract-me-not#29]
This is a simple roadmap for things that would be interesting to add in the future releases:

> Feel free to contribute to this list if ever you wanna see any new feature been added to the extension, perhaps adding some screenshots or code examples can also be really helpful.

- [ ] Icon should change per tab/website, depending on if the url is blocked or not (maybe made this optional via an option on the settings, just to keep the current behavior).
- [ ] Block settings: add an option to prompt websites url & modify it before adding it to blacklist/whitelist (when using the + button).
- [ ] Unblock settings: option to display unblock button on context menu instead of on the page itself (can be handy to hide the button).

**Comment by oatly93** [comment]
Hey man,
great job so far! I'm really loving the extension - as said, I'm using it to overcome an addiction...
I randomized my password so I don't get the option to just unblock websites at will. However, at times I find websites which I hadn't blocked before so I'd love to see the option of blocking a website with just a button without getting into the whole configuration (specially because I can't, it's password protected and like I said, I randomised the pass so I don't know it).

**Comment by dannyk3941** [comment]
This is an awesome extension, thank you so much for making this.

Are you also taking contributors for this project? I would love to help with this.

**Comment by AXeL-dev** [comment]
> This is an awesome extension, thank you so much for making this.

My pleasure!

> Are you also taking contributors for this project? I would love to help with this.

Yeah sure, the project is open source, so you can simply fork it & send a pull request with your modifications once done, as well, if you have anything to discuss you can either create a new issue or contact me directly on my email or discord account.

**Comment by binarynoise** [comment]
>  Block page template: add possibility to customize the template (set customized HTML for example)

Would be cool to have an option to have the blocked URL in a text box that can be copied.  
I have two browser profiles and use this add-on to block the sites that should be opened in the other profile, but it's a bit inconvenient, because I can't copy the URL into the other browser because for some reason the extension IDs don't match...

**Comment by AXeL-dev** [comment]
> > Block page template: add possibility to customize the template (set customized HTML for example)
> 
> Would be cool to have an option to have the blocked URL in a text box that can be copied. I have two browser profiles and use this add-on to block the sites that should be opened in the other profile, but it's a bit inconvenient, because I can't copy the URL into the other browser because for some reason the extension IDs don't match...

:thinking: I'm not sure of which text box that you're talking about? Can you provide a screenshot/video of the issue? Still if you wanna copy your urls between multiple browsers/profiles you can simply use the export/import feature.

**Comment by AXeL-dev** [comment]
Well that makes sense, sure i'll add this feature on the next release (& make it optional).

**Comment by AXeL-dev** [comment]
@binarynoise the feature you asked for was added in version `2.7.0`, you still have to enable it from the blocking settings "Display blocked link" option.

**Comment by binarynoise** [comment]
Looks great, thank you!

**Comment by EftekhariSara** [comment]
This blocker is awesome! I love the wildcard matching option + the ability to just close the tab as one of the options. This has been the most effective method I've come across. So thank you for this amazing blocker!

I think it would be great if there was a way to customise this to happen only for certain websites during a set schedule, and not all of the blacklist.

There are sites I want to block permanently (which I can currently) and others which I want blocked only during the night to facilitate sleep. But if I set a schedule, it all applies to all the blacklist and whitelist.

**Comment by AXeL-dev** [comment]
Hello @EftekhariSara , i agree that it would be great to have multiple black/white lists that can be used per schedule, i'll think about it for the future versions, in the meanwhile 📢  if ever someone is interested to work on this feature i would be happy to review and publish it.
